clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/KafkaInputFormat.java

```diff
@@ -86,21 +89,29 @@ protected Map<PubSubTopicPartition, Long> getLatestOffsets(JobConf config) {
    */
   @Override
   public InputSplit[] getSplits(JobConf job, int numSplits) throws IOException {
-    long maxRecordsPerSplit =
-        job.getLong(KAFKA_INPUT_MAX_RECORDS_PER_MAPPER, DEFAULT_KAFKA_INPUT_MAX_RECORDS_PER_MAPPER);
-    if (maxRecordsPerSplit < 1L) {
+    long kifMapperMultiplicationFactor =
+        job.getLong(KAFKA_MAPPER_MULTIPLICATION_FACTOR, DEFAULT_KAFKA_INPUT_MAPPER_MULTIPLICATION_FACTOR);
+    if (kifMapperMultiplicationFactor < 1L) {
       throw new IllegalArgumentException(
-          "Invalid " + KAFKA_INPUT_MAX_RECORDS_PER_MAPPER + " value [" + maxRecordsPerSplit + "]");
+          "Invalid " + DEFAULT_KAFKA_INPUT_MAPPER_MULTIPLICATION_FACTOR + " value [" + kifMapperMultiplicationFactor
+              + "]");
     }
 
-    return getSplitsByRecordsPerSplit(job, maxRecordsPerSplit);
+    return getSplitsByRecordsPerSplit(job, kifMapperMultiplicationFactor);
   }
 
-  public InputSplit[] getSplitsByRecordsPerSplit(JobConf job, long maxRecordsPerSplit) {
+  public InputSplit[] getSplitsByRecordsPerSplit(JobConf job, long kifMapperMultiplicationFactor) {
     Map<PubSubTopicPartition, Long> latestOffsets = getLatestOffsets(job);
+    long totalEndOffset = latestOffsets.values().stream().mapToLong(Long::longValue).sum();
+    long maxMapperCount = job.getLong(KAFKA_INPUT_MAX_MAPPER_COUNT, DEFAULT_MAX_KIF_MAPPER_COUNT);
+    long mapperCount = latestOffsets.size() * kifMapperMultiplicationFactor;
```

Comment:
Could we please avoid doing arithmetic directly on offsets? This type of code will slow down migration to PubSubPosition where arithmetic ops are not allowed. Rewriting anything other than simple conversions to subtrations is gonging to tough so let's avoid it to begin with

If you still need to perform arithmetic, consider retrieving both the starting and ending offsets, computing the difference explicitly, and then applying any further operations to that difference.